### PR TITLE
40 refactor panelappdbpy and db logic in main

### PR DESCRIPTION
**Pull request by Fr3dMer**
- Added logic and began integration of db with temporary place holders. Will finish off in sprint 5. 
- See issue #40 for further details on what was done

### CODE REVIEW

File: PannelAppDB.py [hunk 1]
```diff
@@ -4,55 +4,107 @@
...
+import pandas as pd
 import src.api as api_module
 import src.cli as cli_module
+import src.database_3.0.py as db
+from datetime import datetime
 import src.JSON_parsing as parser_obj
-import sys
-
 
 def main():
```
Comment (AmyGrimwood) [inline PannelAppDB.py:15]: There should be 2 empty lines above function definition

File: src/api.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def value_checker(self,pannel_id=None,version=None,rcode=None):
     def version_check(self,query_version,true_version):
 
         if not (type(query_version) == float and type(true_version) == float):
-            raise SyntaxError("Verions are not floats, cannot compare versions")
+            raise SyntaxError("Versions are not floats, cannot compare versions")
```
Comment (AmyGrimwood) [inline src/api.py:60]: Line longer than 79 characters (PEP8)

File: src/api.py [hunk 3]
```diff
@@ -75,6 +75,24 @@ def check_internet(self,url = 'http://google.com'):
         else:
             return True
 
+    # Extracts version from raw_data, calls api to ensure most recent GMS panel,
```
Comment (AmyGrimwood) [inline src/api.py:78]: Line longer than 79 characters

File: src/cli.py [hunk 4]
```diff
@@ -28,3 +28,18 @@ def __init__(self,raw_args):
...
         self.args = parser.parse_args(raw_args)
 
+        # Check if any panel id entered
+        if (type(self.args.panel_id) == int or type(self.args.rcode == int)):
+            self.return_panel_info = True
+        else:
+            self.return_panel_info = False
+
+        # Check if any patient details enteres
+        if (type(self.args.patientID) == str or type(self.args.sampletID) == str):
```
Comment (AmyGrimwood) [inline src/cli.py:38]: Line longer than 79 chars

**Review by AmyGrimwood — CHANGES_REQUESTED** [review]
Theres a few lines that might have PEP8 issues. Please let me know if you agree

File: PannelAppDB.py (hunk 1 above)
Comment (Fr3dMer) [reply PannelAppDB.py:15]: Good catch, resolved.

File: src/api.py (hunk 2 above)
Comment (Fr3dMer) [reply src/api.py:60]: Entire class does not conform to pep-8, will tackle entire module in sprint 5

File: src/api.py (hunk 3 above)
Comment (Fr3dMer) [reply src/api.py:78]: Entire class does not conform to pep-8, will tackle entire module in sprint 5

File: src/cli.py (hunk 4 above)
Comment (Fr3dMer) [reply src/cli.py:38]: Entire class does not conform to pep-8, will tackle entire module in sprint 5

**Comment by Fr3dMer** [comment]
Agreed those are all PEP8 violations, modules api.py and cli.py have many non-conformities. Think they will need thorough refactoring, better to save for sprint 5?

File: PannelAppDB.py [hunk 5]
```diff
@@ -4,55 +4,107 @@
 Author        : Freddie Mercer
 ****************************************************************************"""
```
Comment (AmyGrimwood) [inline PannelAppDB.py:6]: Need doc string at top of script/module

File: PannelAppDB.py [hunk 6]
```diff
@@ -4,55 +4,107 @@
...
+import pandas as pd
 import src.api as api_module
 import src.cli as cli_module
+import src.database_3.0.py as db
+from datetime import datetime
 import src.JSON_parsing as parser_obj
-import sys
 
 
 def main():
```
Comment (AmyGrimwood) [inline PannelAppDB.py:16]: Need doc string for function

File: src/api.py [hunk 7]
```diff
@@ -57,7 +57,7 @@ def value_checker(self,pannel_id=None,version=None,rcode=None):
     def version_check(self,query_version,true_version):
```
Comment (AmyGrimwood) [inline src/api.py:57]: Need docstring for function

**Review by AmyGrimwood — APPROVED** [review]
As discussed, module isnt compliant with PEP8 but will be addressed in sprint 5

**Pull request merged**